### DISCUSSION TITLE
How do i disable all action rows in a slash command's reply function

### DISCUSSION THREAD

**Discussion by JustRed23** [discord-jda/JDA#1789] [Questions and Help] [answered]
So i have a command (/test) and i want to display text ("Is this working?") with a simple yes and no button. This part already works.
But now i want to delete those buttons when one of them is clicked so you can not click them again. How do i do that?

The image below is the message:
![image](https://user-images.githubusercontent.com/21986809/130679741-7fbe4119-a047-448f-b41b-305aa1050789.png)

**Comment by gpluscb** [comment, marked as answer]
Using `setActionRows` with no arguments on the edit action should do the trick.

**Reply by JustRed23** [reply]
Thank you. This did the trick

**Reply by ThatTaroEXE** [reply]
It replies "Cannot have empty row!" for me,

But "event.editSelectMenu(null).queue();" do it for me.

**Comment by edusrossetto** [comment]
01/04/2022 Jda 5.0
Using `event.getMessage().editMessageEmbeds().setActionRow();` will trigger
```
[JDA MainWS-ReadThread] INFO JDA - Finished Loading!
[JDA MainWS-ReadThread] ERROR JDA - One of the EventListeners had an uncaught exception
java.lang.IllegalArgumentException: Cannot have empty row!
```

**Reply by RedcodesDev** [reply]
Use `setActionRows()` instead.

**Reply by edusrossetto** [reply]
Now it returns
`[ForkJoinPool.commonPool-worker-1] ERROR RestAction - RestAction queue returned failure: [ErrorResponseException] 50006: Cannot send an empty message`
`event.getMessage().editMessageEmbeds().setActionRows().queue();`

**Reply by pp0236** [reply]
> Now it returns `[ForkJoinPool.commonPool-worker-1] ERROR RestAction - RestAction queue returned failure: [ErrorResponseException] 50006: Cannot send an empty message` `event.getMessage().editMessageEmbeds().setActionRows().queue();`

```
MessageEmbed original = event.getMessage().getEmbeds().get(0);
event.getMessage().editMessageEmbeds(original).setActionRows().queue();
```
do something like this (code from above is on jda 4.3 not 5.0

**Reply by SvenWollinger** [reply]
Was there an update to this? Im on 5.0.0-beta.3 and when editing a Message i do not have ```setActionRows```, just setActionRow which throws "Cannot have empty row".
There must be a way to clear a message of Buttons, no?

**Reply by MinnDevelopment** [reply]
Its now called setComponents

**Comment by DidsheMa** [comment]
event.editComponents().queue(); 

for fellow confused people from 2024 having tried all these unsuccessfully 🥴

**Reply by pp0236** [reply]
fr
